integrationtests/src/test/java/org/neo4j/procedure/ProcedureIT.java

```diff
@@ -924,11 +924,15 @@ public void shouldBeAbleToSpawnThreadsCreatingTransactionInProcedures() throws T
             threads[i].join();
         }
 
-        Result result = db.execute( "MATCH () RETURN count(*) as n" );
-        assertThat( result.hasNext(), equalTo( true ) );
-        while ( result.hasNext() )
+        Result result;
+        try
         {
-            assertThat( result.next().get( "n" ), equalTo( (long) numThreads ) );
+            result = getAndAssertNodeCountResult( numThreads );
```

Comment:
What kind of timing issues would render this to give different results? Especially since we join on all threads before I don't really understand. What did the failures look before and could you locally verify if this reduced the chance of failure?

Reply:
First of all, I could not reproduce it locally but it seems to happen regularly on TC. The failure always gets 9 nodes instead of 10 which led me to speculate that although all threads are joined, at least one transaction hasn't been completely finished in the background by the time the count(*) transaction is opened. Sadly this is only a theory at this point, thus my not-harmful "fix"

Reply:
I can't see how the transaction should not be finished after we join on the Thread. For me it looks rather like some concurrency between the Threads, for which this fix would not help.

I agree it's not harmful. But for me it's not obvious it helps at all. And it case it doesn't we should not merge this. It could also be that there is an underlying real issue and not the test is badly designed.

Reply:
I am open to suggestions :-)